element_array_ephys/export/nwb.py

```diff
@@ -0,0 +1,153 @@
+                        if ephys_device_name in nwbfile.devices
+                        else nwbfile.create_device(name=ephys_device_name))
+
+        electrode_group = nwbfile.create_electrode_group(
+            name=f'{electrode_config["probe"]} {electrode_config["electrode_config_name"]}',
+            description=json.dumps(electrode_config, default=str),
+            device=ephys_device,
+            location=insert_location)
+
+        electrode_query = (probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode
+                           & electrode_config)
+        for electrode in electrode_query.fetch(as_dict=True):
+            nwbfile.add_electrode(
+                id=electrode['electrode'], group=electrode_group,
+                filtering='', imp=-1.,
+                x=np.nan, y=np.nan, z=np.nan,
+                rel_x=electrode['x_coord'], rel_y=electrode['y_coord'], rel_z=np.nan,
+                shank=electrode['shank'], shank_col=electrode['shank_col'], shank_row=electrode['shank_row'],
+                location=electrode_group.location)
+
+        # ---- Units ----
+        unit_query = clustering_query @ ephys.CuratedClustering.Unit
+        for unit in unit_query.fetch(as_dict=True):
+            waveform_mean = waveform_std = np.full(1, np.nan)
+            if ephys.WaveformSet & curated_clustering_key & {'unit': unit['unit']}:
+                waveform_mean = (ephys.WaveformSet.PeakWaveform
+                                 & curated_clustering_key
+                                 & {'unit': unit['unit']}).fetch1('peak_electrode_waveform')
+                # compute waveform's STD across spikes from the peak electrode
+                electrode_waveforms = (ephys.WaveformSet.Waveform & curated_clustering_key
+                                       & {'unit': unit['unit']}
+                                       & {'electrode': unit['electrode']}).fetch1('waveforms')
+                if electrode_waveforms is not None:
+                    waveform_std = np.std(electrode_waveforms, axis=0)
+
+            nwbfile.add_unit(id=unit['unit'],
+                             electrodes=np.where(nwbfile.electrodes.id.data == int(unit['electrode']))[0],
+                             electrode_group=electrode_group,
+                             sampling_rate=unit['sampling_rate'],
+                             cluster_quality_label=unit['cluster_quality_label'],
+                             spike_times=unit['spike_times'],
+                             spike_sites=unit['spike_sites'],
+                             spike_count=unit['spike_count'],
+                             spike_depths=unit['spike_depths'],
+                             waveform_mean=waveform_mean,
+                             waveform_sd=waveform_std)
+
+        # ---- Local Field Potential ----
+        if ephys.LFP.Electrode & curated_clustering_key:
+            nwb_lfp = pynwb.ecephys.LFP(name=f'probe_{electrode_config["probe"]} - LFP')
```

Comment:
I don't know if you need to name this according to a probe. The underlying `ElectricalSeries` will have an electrodes attribute that points to a subregion of the electrodes table